### ISSUE TITLE
Allow user to control content type in newly-created GoogleStorageResource

### ISSUE THREAD

**Issue by dzou** [spring-cloud/spring-cloud-gcp#1603]
Allow the user to set the Content Type when GoogleStorageResource is created. This is followup to https://github.com/spring-cloud/spring-cloud-gcp/pull/1554#pullrequestreview-218983348

**Comment by meltsufin** [comment]
Actually related to #861. @dzou What do you think of that approach? Should it be made easier?

**Comment by dzou** [comment]
> Actually related to #861. @dzou What do you think of that approach? Should it be made easier?

Yeah, I think it could be made easier; I am not a big fan of that approach simply because you have to know the Storage API pretty well to come up with that solution (or be lucky that you found it in the docs).

We could come up with some way of creating/modifying the metadata more easily; seems like there are [theoretically many settings a user might want to set](https://googleapis.github.io/google-cloud-java/google-cloud-clients/apidocs/com/google/cloud/storage/BlobInfo.Builder.html).
